test/integration/scheduler_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/scheduler"
+	schedapi "k8s.io/kubernetes/pkg/scheduler/apis/config"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	testutils "k8s.io/kubernetes/test/integration/util"
+
+	"sigs.k8s.io/scheduler-plugins/pkg/qos"
+)
+
+func TestSchedulerCreation(t *testing.T) {
+	tests := []struct {
+		name            string
+		registry        framework.Registry
+		plugins         *schedapi.Plugins
+		expectedPlugins map[string][]schedapi.Plugin
+	}{
+		{
+			name:     "QoS QueueSort plugin",
+			registry: framework.Registry{qos.Name: qos.New},
+			plugins: &schedapi.Plugins{
+				QueueSort: &schedapi.PluginSet{
+					Enabled: []schedapi.Plugin{
+						{Name: qos.Name},
+					},
+					Disabled: []schedapi.Plugin{
+						{Name: "*"},
+					},
+				},
+			},
+			expectedPlugins: map[string][]schedapi.Plugin{
+				"QueueSortPlugin": {{Name: qos.Name}},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			profile := schedapi.KubeSchedulerProfile{
+				SchedulerName: v1.DefaultSchedulerName,
+				Plugins:       tt.plugins,
+			}
+			testCtx := testutils.InitTestSchedulerWithOptions(
+				t,
+				testutils.InitTestMaster(t, "sched-creation", nil),
+				false,
+				nil,
+				0,
+				scheduler.WithProfiles(profile),
+				scheduler.WithFrameworkOutOfTreeRegistry(tt.registry),
+			)
+			defer testutils.CleanupTest(t, testCtx)
+
+			schedPlugins := testCtx.Scheduler.Profiles[v1.DefaultSchedulerName].ListPlugins()
+			// We only check the particular extension points defined in each test case.
+			for extension, expected := range tt.expectedPlugins {
```

Comment:
What is the point of this test?
Tests for scheduler creation are already in k/k.

Reply:
It's aimed to ensure the out-of-tree plugin works well with default ones. Especially for some plugin which should be global unique, e.g., QueueSort plugin.

Reply:
sg